### PR TITLE
docs(router): Updated routing examples to use routing module

### PR DESCRIPTION
**Pull request by brandonroberts**
All examples updating to use separate "routing modules"
Routing tutorial simplified and prose updated
Router developer guide examples and prose updated

### CODE REVIEW

**Comment by brandonroberts** [comment]
@wardbell @johnpapa this PR is ready for review. All the samples have been updated to use `Routing Modules`. I didn't change any of the prose in the `NgModule` chapter

File: public/docs/ts/latest/tutorial/toh-pt5.jade [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 block includes
   include ../_util-fns
-  - var _appRoutingTsVsAppComp = 'app.routing.ts'
+  - var _appRoutingTsVsAppComp = 'app.module.ts'
```
Comment (johnpapa) [inline public/docs/ts/latest/tutorial/toh-pt5.jade]: what does this jade syntax do? If its code, we dont use underscore prefixes by convention. If its styling, ignore this :)

File: public/docs/ts/latest/tutorial/toh-pt6.jade [hunk 2]
```diff
@@ -229,7 +229,7 @@ block get-heroes-details
   ## Update hero details
 
   We can edit a hero's name already in the hero detail view. Go ahead and try
-  it. As we type, the hero name is updated in the view heading. 
+  it. As we type, the hero name is updated in the view heading.
```
Comment (johnpapa) [inline public/docs/ts/latest/tutorial/toh-pt6.jade]: @wardbell mentioned to me a few days ago that we are switching from the "we" to the "you" in our prose. If so, this needs to switch. But check with Ward.

File: public/docs/ts/latest/tutorial/toh-pt5.jade (hunk 1 above)
Comment (chalin) [inline public/docs/ts/latest/tutorial/toh-pt5.jade]: Local Jade variable names used to render TS and Dart prose from the same source. Such variable names do not appear in the rendered HTML.

Comment (johnpapa) [inline public/docs/ts/latest/tutorial/toh-pt5.jade]: perfect. thanks

File: public/docs/ts/latest/tutorial/toh-pt6.jade (hunk 2 above)
Comment (wardbell) [inline public/docs/ts/latest/tutorial/toh-pt6.jade]: We are switching ot _you_. But this switch is just one part of a deliberate sweep of the docs leading to conformance with the Google docs standards. No need for Brandon to take that on for _this_ particular revision.

_After_  a doc has been converted, _then_ our revisions must conform.

**Comment by brandonroberts** [comment]
@wardbell @johnpapa I updated the ngmodule guide and ngmodule faq cookbook with sections for routing modules. Its review for complete review

**Comment by wardbell** [comment]
TODO: incorporate feed back in issue #2527

**Pull request merged**